client/lib/feed-stream-store/post-key.js

```diff
@@ -0,0 +1,34 @@
+export function keyForPost( post ) {
+	if ( post.feed_ID && post.feed_item_ID ) {
+		return {
+			feedId: post.feed_ID,
+			postId: post.feed_item_ID
+		};
+	}
+	if ( post.is_external ) {
+		return {
+			feedId: post.feed_ID,
+			postId: post.ID
+		};
+	}
+	return {
+		blogId: post.site_ID,
+		postId: post.ID
+	};
+}
+
+export function keysAreEqual( a, b ) {
```

Comment:
hmm, i think this is actually busted. doesn't deal with gaps. depending on usage, might need to deal with recs blocks and combos too. I don't really consider those postKeys (even gaps are questionable, but at least they're part of the stream data), but item descriptors... 

one nice thing about isEqual is that it just worked with things like gaps, but it requires that the objects all agree on a shape, which can lead to subtle bugs.